Add put block

BrianMcBrayer opened the pull request:
* Added Base64 helper converter
* Added Guard to check for string size and Base64 encoding
* Implemented PutBlock request and response

tarwn commented:
Code Review Notes:

* Basic.Azure.Storage/BlobserviceClient.cs - the comments are Copy/Pasta, they still have the text from the PutBlockBlob method
*  Basic.Azure.Storage/BlobserviceClient.cs - PutBlockAsync - the pattern for the other async calls has been to await the async method, not the return variable
* I don't see how PutBlock_DifferentLengthBlockIds_ThrowsArgumentException is working? There aren't any guard checks and we don't track state to compare two block id's across separate calls? (Maybe i'm just missing something)

BrianMcBrayer commented:
Good catch on all points. 
* I removed the comments since we are not primarily using XML comments. 
* Updated the await/async pattern.
* Correct on the ArgumentException. It was passing because the wrong-length ID was not Base64 encoded. Updated the test and verified results.
